File: health_advanced/HealthDashboard/Controllers/HDExerciseTypeViewController.m
//
//  HDExerciseTypeViewController.m
//  HealthDashboard
//
//  Created by zhang, haizi on 2026/3/21.
//

#import "HDExerciseTypeViewController.h"
#import "HDExerciseTypeView.h"
#import "HDExerciseSettingViewController.h"
#import "../Models/HDHealthDataModel.h"

@interface HDExerciseTypeViewController ()
@property (nonatomic, strong) UIScrollView *scrollView;
@property (nonatomic, strong) HDExerciseTypeView *targetRunView;
@property (nonatomic, strong) HDExerciseTypeView *freeRunView;
@end

@implementation HDExerciseTypeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupUI];
    [self applyTheme];
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    [self applyTheme];
}

- (void)setupUI {
    self.view.backgroundColor = UIColor.systemBackgroundColor;
    self.title = @"选择运动类型";
    
    // ScrollView
    _scrollView = [UIScrollView new];
    _scrollView.showsVerticalScrollIndicator = NO;
    [self.view addSubview:_scrollView];
    
    // 目标跑卡片
    _targetRunView = [HDExerciseTypeView new];
    _targetRunView.exerciseType = 0;
    _targetRunView.titleLabel.text = @"🏃 目标跑";
    _targetRunView.descriptionLabel.text = @"设置距离和时间目标\n开始计时运动";
    [_targetRunView.startButton addTarget:self action:@selector(startTargetRun) forControlEvents:UIControlEventTouchUpInside];
    [_scrollView addSubview:_targetRunView];
    
    // 自由跑卡片
    _freeRunView = [HDExerciseTypeView new];
    _freeRunView.exerciseType = 1;
    _freeRunView.titleLabel.text = @"🏃 自由跑";
    _freeRunView.descriptionLabel.text = @"无目标限制\n尽情享受运动";
    [_freeRunView.startButton addTarget:self action:@selector(startFreeRun) forControlEvents:UIControlEventTouchUpInside];
    [_scrollView addSubview:_freeRunView];
    
    // 布局
    [self layoutUI];
}

- (void)layoutUI {
    CGFloat padding = 16;
    CGFloat width = self.view.bounds.size.width - padding * 2;
    
    _scrollView.frame = self.view.bounds;
    
    _targetRunView.frame = CGRectMake(padding, padding, width, 180);
    _freeRunView.frame = CGRectMake(padding, padding * 2 + 180, width, 180);
    
    _scrollView.contentSize = CGSizeMake(self.view.bounds.size.width, padding * 3 + 360);
}

- (void)startTargetRun {
    HDExerciseSettingViewController *vc = [HDExerciseSettingViewController new];
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)startFreeRun {
    // TODO: 跳转到计时器页面（自由跑）
    UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"自由跑" message:@"功能开发中..." preferredStyle:UIAlertControllerStyleAlert];
    [alert addAction:[UIAlertAction actionWithTitle:@"确定" style:UIAlertActionStyleDefault handler:nil]];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)applyTheme {
    self.view.backgroundColor = UIColor.systemBackgroundColor;
    [_targetRunView applyTheme];
    [_freeRunView applyTheme];
}

@end

File: health_advanced_swift_spec/HealthDashboard/Controllers/HDExerciseTypeViewController.swift
//
//  HDExerciseTypeViewController.swift
//  HealthDashboard - 运动类型选择
//
//  Created by zhang, haizi on 2026/3/22.
//

import UIKit
import Combine
import Combine

// MARK: - ViewController

@MainActor
class HDExerciseTypeViewController: UIViewController {

    // MARK: - UI

    private let scrollView = UIScrollView()
    private var targetRunView: HDExerciseTypeView!
    private var freeRunView: HDExerciseTypeView!
    private var cancellables = Set<AnyCancellable>()

    // MARK: - Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        setupUI()
        applyTheme()
        // 订阅主题变化
        HDHealthDataModel.shared.$isDarkMode
            .receive(on: DispatchQueue.main)
            .sink { [weak self] _ in self?.applyTheme() }
            .store(in: &cancellables)
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        applyTheme()
    }

    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        layoutUI()
    }

    // MARK: - UI Setup

    private func setupUI() {
        title = "选择运动类型"

        scrollView.showsVerticalScrollIndicator = false
        view.addSubview(scrollView)

        // 目标跑卡片
        targetRunView = HDExerciseTypeView()
        targetRunView.exerciseType = 0
        targetRunView.titleLabel.text = "🏃 目标跑"
        targetRunView.descriptionLabel.text = "设置距离和时间目标\n开始计时运动"
        targetRunView.startButton.addTarget(self, action: #selector(startTargetRun), for: .touchUpInside)
        scrollView.addSubview(targetRunView)

        // 自由跑卡片
        freeRunView = HDExerciseTypeView()
        freeRunView.exerciseType = 1
        freeRunView.titleLabel.text = "🏃 自由跑"
        freeRunView.descriptionLabel.text = "无目标限制\n尽情享受运动"
        freeRunView.startButton.addTarget(self, action: #selector(startFreeRun), for: .touchUpInside)
        scrollView.addSubview(freeRunView)
    }

    private func layoutUI() {
        let padding: CGFloat = 16
        let width = view.bounds.width - padding * 2

        scrollView.frame = view.bounds
        targetRunView.frame = CGRect(x: padding, y: padding, width: width, height: 180)
        freeRunView.frame = CGRect(x: padding, y: padding * 2 + 180, width: width, height: 180)
        scrollView.contentSize = CGSize(width: view.bounds.width, height: padding * 3 + 360)
    }

    // MARK: - Actions

    @objc private func startTargetRun() {
        let vc = HDExerciseSettingViewController()
        navigationController?.pushViewController(vc, animated: true)
    }

    @objc private func startFreeRun() {
        let alert = UIAlertController(title: "自由跑", message: "功能开发中...", preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "确定", style: .default))
        present(alert, animated: true)
    }

    // MARK: - Theme

    @objc func applyTheme() {
        view.backgroundColor = .systemBackground
        targetRunView?.applyTheme()
        freeRunView?.applyTheme()
    }
}
